ReactiveCocoa/UIKit/iOS/UIKeyboard.swift

```diff
@@ -37,19 +68,31 @@ public struct KeyboardChangeContext {
 		return base[UIKeyboardIsLocalUserInfoKey] as! Bool
 	}
 
-	fileprivate init(_ userInfo: [AnyHashable: Any]) {
+	fileprivate init(userInfo: [AnyHashable: Any], event: KeyboardEvent) {
 		base = userInfo
+		self.event = event
 	}
 }
 
 extension Reactive where Base: NotificationCenter {
+	/// Create a `Signal` that notifies whenever the system keyboard announces specified events.
+	///
+	/// - returns: A `Signal` that emits the context of system keyboard events.
+	public func keyboard(_ events: KeyboardEvent...) -> Signal<KeyboardChangeContext, NoError> {
```

Comment:
You should probably be required to pass at least one event.

We can do that like so:

```swift
func keyboard(_ first: KeyboardEvent, _ rest: KeyboardEvent…) -> Signal<KeyboardChangeContext, NoError> {
```